tests/unit/modules/test_chroot.py

```diff
@@ -0,0 +1,281 @@
+                          'test.ping')
+
+    @patch('salt.modules.chroot.exist')
+    @patch('tempfile.mkdtemp')
+    def test_call_fails_untar(self, mkdtemp, exist):
+        '''
+        Test execution of Salt functions in chroot.
+        '''
+        # Fail the tar command
+        exist.return_value = True
+        mkdtemp.return_value = '/chroot/tmp01'
+        utils_mock = {
+            'thin.gen_thin': MagicMock(return_value='/salt-thin.tgz'),
+            'files.rm_rf': MagicMock(),
+        }
+        salt_mock = {
+            'cmd.run': MagicMock(return_value='Error'),
+            'config.option': MagicMock(),
+        }
+        with patch.dict(chroot.__utils__, utils_mock), \
+                patch.dict(chroot.__salt__, salt_mock):
+            self.assertEqual(chroot.call('/chroot', 'test.ping'), {
+                'result': False,
+                'comment': 'Error'
+            })
+            utils_mock['thin.gen_thin'].assert_called_once()
+            salt_mock['config.option'].assert_called()
+            salt_mock['cmd.run'].assert_called_once()
+            utils_mock['files.rm_rf'].assert_called_once()
+
+    @patch('salt.modules.chroot.exist')
+    @patch('tempfile.mkdtemp')
+    def test_call_fails_salt_thin(self, mkdtemp, exist):
+        '''
+        Test execution of Salt functions in chroot.
+        '''
+        # Fail the inner command
+        exist.return_value = True
+        mkdtemp.return_value = '/chroot/tmp01'
+        utils_mock = {
+            'thin.gen_thin': MagicMock(return_value='/salt-thin.tgz'),
+            'files.rm_rf': MagicMock(),
+            'json.find_json': MagicMock(return_value={'return': {}})
+        }
+        salt_mock = {
+            'cmd.run': MagicMock(return_value=''),
+            'config.option': MagicMock(),
+            'cmd.run_chroot': MagicMock(return_value={
+                'retcode': 1,
+                'stderr': 'Error',
+            }),
+        }
+        with patch.dict(chroot.__utils__, utils_mock), \
+                patch.dict(chroot.__salt__, salt_mock):
+            self.assertEqual(chroot.call('/chroot', 'test.ping'), {
+                'result': False,
+                'comment': "Can't parse container command output"
+            })
+            utils_mock['thin.gen_thin'].assert_called_once()
+            salt_mock['config.option'].assert_called()
+            salt_mock['cmd.run'].assert_called_once()
+            salt_mock['cmd.run_chroot'].assert_called_with(
+                '/chroot',
+                ['python{}'.format(sys.version_info[0]), '/tmp01/salt-call',
+                 '--metadata', '--local',
+                 '--log-file', '/tmp01/log', '--cachedir', '/tmp01/cache',
+                 '--out', 'json', '-l', 'quiet', '--', 'test.ping'])
+            utils_mock['files.rm_rf'].assert_called_once()
+
+    @patch('salt.modules.chroot.exist')
+    @patch('tempfile.mkdtemp')
+    def test_call_success(self, mkdtemp, exist):
```

Comment:
This test is failing on windows:
```Traceback (most recent call last):
  File "c:\users\admini~1\appdata\local\temp\kitchen\testing\.nox\runtests-parametrized-2-7-coverage-true-crypto-none-transport-zeromq\lib\site-packages\mock\mock.py", line 1330, in patched
    return func(*args, **keywargs)
  File "c:\users\admini~1\appdata\local\temp\kitchen\testing\tests\unit\modules\test_chroot.py", line 198, in test_call_success
    '--out', 'json', '-l', 'quiet', '--', 'test.ping'])
  File "c:\users\admini~1\appdata\local\temp\kitchen\testing\.nox\runtests-parametrized-2-7-coverage-true-crypto-none-transport-zeromq\lib\site-packages\mock\mock.py", line 944, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "c:\users\admini~1\appdata\local\temp\kitchen\testing\.nox\runtests-parametrized-2-7-coverage-true-crypto-none-transport-zeromq\lib\site-packages\six.py", line 737, in raise_from
    raise value
AssertionError: expected call not found.
Expected: mock(u'/chroot', [u'python2', u'/tmp01/salt-call', u'--metadata', u'--local', u'--log-file', u'/tmp01/log', u'--cachedir', u'/tmp01/cache', u'--out', u'json', u'-l', u'quiet', u'--', u'test.ping'])
Actual: mock(u'/chroot', ['python2', '\\tmp01\\salt-call', '--metadata', '--local', '--log-file', '\\tmp01\\log', '--cachedir', '\\tmp01\\cache', '--out', 'json', '-l', 'quiet', '--', 'test.ping'])
```

Reply:
I don't imagine the `chroot` module works on windows so we can probably just add a `@skipIf(salt.utils.platform.is_windows())` to the top of the test class.

Reply:
Done